test/service-worker_test.js

```diff
@@ -64,21 +64,36 @@ suite('service-worker', () => {
   suite('generateServiceWorker()', () => {
 
     test('should throw when options are not provided', () => {
-      assert.throws(() => {
-        serviceWorker.generateServiceWorker();
-      }, Error.AssertionError, '`project` & `buildRoot` options are required');
+      return serviceWorker.generateServiceWorker()
+        .then(
+          () => { throw new Error('generateServiceWorker() resolved, expected rejection!'); },
```

Comment:
instead of throwing, you can: `assert.fail('generateServiceWorker() resolved, expected rejection!')`